std/typecons.d

```diff
@@ -422,19 +422,31 @@ template Tuple(Specs...)
         U u = U.init;
         T t = u;
     }));
+
     /+ Helper for partial instanciation +/
     template isBuildableFrom(U)
     {
         enum isBuildableFrom(T) = isBuildable!(T, U);
     }
 
+    /+ Determine whether or not all given types in T are the same type +/
+    template areAllSame(T...)
```

Comment:
I thought about this, but ultimately, I think it's better without. The biggest issue is that the element type wouldn't be an lvalue like it is with the current implementation.